tensorflow_zero_out/cc/kernels/lsh_matmul.cc

```diff
@@ -0,0 +1,221 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/core/framework/op_kernel.h"
+
+using namespace tensorflow;
+
+typedef Eigen::ThreadPoolDevice CPUDevice;
+
+template <typename Device, typename T, bool USE_CUBLAS>
+
+class MklMatMulOp : public OpKernel {
+ public:
+  explicit MklMatMulOp(OpKernelConstruction* ctx) : OpKernel(ctx) {
+    OP_REQUIRES_OK(ctx, ctx->GetAttr("transpose_a", &transpose_a_));
+    OP_REQUIRES_OK(ctx, ctx->GetAttr("transpose_b", &transpose_b_));
+  }
+
+#define PADDING 4
+#define BISINIT 0
+#define BINDEX 1
+#define BCOUNT 2
+//TODO: fill the values below
+#define BUCKETSIZE 
+#define _L
+#define _numhashes 
+#define _noOfNodes 
+
+  int Bucket_add(int id, int l, int idx, int L, int RangePow, int BUCKETSIZE,
```

Comment:
RangePow is only needed for HashFunction=3. You can ignore it for simplicity since you are focusing on SRP. It seems you are trying to implement this: int LSH::add (int tableId, int indices, int id), is that right?